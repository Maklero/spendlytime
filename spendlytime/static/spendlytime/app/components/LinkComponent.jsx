import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function LinkComponent({ children, link }){
    return(
        <Link to={link ? link : "#"}>
            {children}
        </Link>
    );
}

LinkComponent.propTypes = {
    children: PropTypes.any,
    link: PropTypes.string
}

export default LinkComponent;