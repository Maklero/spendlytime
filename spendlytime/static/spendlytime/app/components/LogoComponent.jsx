import React from 'react';

export default function Logo(){
    return(
        <div className="logo">
            <h1 className="logo__text">Spendlytime.com</h1>
        </div>
    );
}