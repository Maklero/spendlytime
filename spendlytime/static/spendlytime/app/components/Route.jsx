import React from 'react';
import { Route as BaseRoute, Redirect } from 'react-router-dom';

export default function Route(props){
    const { authRequired } = props;

    if (authRequired) {
        if (window.__initialData){
            if(window.__initialData.isAuthenticated){
                return <BaseRoute {...props}/>;
            } else {
                return <Redirect to="auth/login" />;
            }
        } else {
            return <Redirect to="auth/login" />;
        }
    } else {
        return <BaseRoute {...props}/>;
    }
}