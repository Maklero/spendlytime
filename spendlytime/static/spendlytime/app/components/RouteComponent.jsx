import React from 'react';
import { Route as BaseRoute, Redirect } from 'react-router-dom';

export default function RouteComponent(props){
    const { authRequired } = props;

    if (authRequired) {
        if (window.__initialData){
            if(window.__initialData.isAuthenticated){
                return <BaseRoute {...props}/>;
            } else {
                return <Redirect to="auth/login" />;
            }
        } else {
            return <BaseRoute {...props}/>;
        }
    } else {
        return <BaseRoute {...props}/>;
    }
}