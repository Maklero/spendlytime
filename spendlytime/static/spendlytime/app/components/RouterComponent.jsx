import React from "react";
import PropTypes from "prop-types";
import { BrowserRouter, Switch } from "react-router-dom";
import { RouteComponent as Route } from '../components';

import styled from 'styled-components';

export const Container = styled.div`
    width: calc(100% - 300px);
    height: 100vh;
    float: left;
`;

function RouterComponent(props) {
    const { routes } = props;
    const routesItems = routes.map((route) => (
        <Route
            key={route.path}
            path={route.path}
            component={route.component}
            exact={route.exact ? true : false}
            authRequired={route.authRequired ? true : false}
        />
    ));

    return (
        <BrowserRouter>
            {props.children}
            <Switch>
                <Container>
                    {routesItems}
                </Container>
            </Switch>
        </BrowserRouter>
    );
}

RouterComponent.propTypes = {
    routes: PropTypes.array.isRequired,
};

export default RouterComponent;
