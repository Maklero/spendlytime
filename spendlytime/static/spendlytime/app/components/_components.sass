.logo
    width: 280px
    height: 38px
    display: inline-block

.logo__text
    color: #00D856
    font-size: 2.1rem;
    font-weight: bold