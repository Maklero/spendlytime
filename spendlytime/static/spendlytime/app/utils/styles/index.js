import { colors } from './colors';
import { device } from './media';
import { GlobalStyles } from './global';

export {
    colors,
    device,
    GlobalStyles
}