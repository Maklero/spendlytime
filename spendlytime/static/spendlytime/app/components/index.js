import LinkComponent from './LinkComponent';
import RouteComponent from './RouteComponent';
import RouterComponent from './RouterComponent';
import SidebarComponent from './SidebarComponent/SidebarComponent';
import LoadingComponent from './LoadingComponent/LoadingComponent';
import TraceItemComponent from './TraceItemComponent/TraceItemComponent';

export {
    LinkComponent,
    RouteComponent,
    RouterComponent,
    SidebarComponent,
    LoadingComponent,
    TraceItemComponent
}