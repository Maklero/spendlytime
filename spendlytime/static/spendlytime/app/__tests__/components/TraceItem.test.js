import React from "react";
import { create } from "react-test-renderer";

import { Provide } from "../../utils/tests";
import { TraceItemComponent } from "../../components";

describe("Trace item component", () => {
    test("Matches the snapshot", () => {
        const traceitem = create(Provide(TraceItemComponent, {"url": "test", "time": "0"}));
        expect(traceitem.toJSON()).toMatchSnapshot();
    });
});
