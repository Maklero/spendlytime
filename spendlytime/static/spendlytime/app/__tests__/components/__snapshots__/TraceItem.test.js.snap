// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trace item component Matches the snapshot 1`] = `
<div
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="sc-fzoXzr ebpenJ"
  >
    <p
      className="sc-fzpjYC dUbwPq"
    >
      test
    </p>
    <p
      className="sc-fznxsB gPUzsl"
    >
      0
    </p>
  </div>
</div>
`;
